New system table zookeeper connection (#45245)
ClickHouse/ClickHouse: docs/en/operations/system-tables/zookeeper_connection.md

---

> slug: /zh/operations/system-tables/zookeeper_connection

Reviewer: ```suggestion
slug: /en/operations/system-tables/zookeeper_connection
```

---
slug: /zh/operations/system-tables/zookeeper_connection
---
#zookeeper_connection

> This table does not exist if ClickHouse Cluster does not config zookeeper. The 'system.zookeeper_connection' table will shows current clickhouse-cluster connected zookeepers info.
> 
> each rows will shows the node information of a zk cluster connected to.

Reviewer: ```suggestion
This table does not exist if ZooKeeper is not configured. The 'system.zookeeper_connection' table shows current connections to ZooKeeper (including auxiliary ZooKeepers). Each row shows information about one connection.
```

Columns:

> -   `name` ([String](../../sql-reference/data-types/string.md)) — zookeeper cluster name.
> -   `host` ([String](../../sql-reference/data-types/string.md)) — connected zookeeper node host.
> -   `port` ([String](../../sql-reference/data-types/string.md)) — connected zookeeper node port.
> -   `index` ([UInt8](../../sql-reference/data-types/int-uint.md)) — connected zookeeper node index in zk cluster config.
> -   `connected_time` ([String](../../sql-reference/data-types/string.md)) — when establish the connection to this zk node.
> -   `is_expired` ([UInt8](../../sql-reference/data-types/int-uint.md)) — Does current connection expired.
> -   `keeper_api_version` ([String](../../sql-reference/data-types/string.md)) — shows keeper api version.
> -   `client_id` ([UInt64](../../sql-reference/data-types/int-uint.md)) — zookeeper connection session id.

Reviewer: ```suggestion
-   `name` ([String](../../sql-reference/data-types/string.md)) — ZooKeeper cluster's name.
-   `host` ([String](../../sql-reference/data-types/string.md)) — The hostname/IP of the ZooKeeper node that ClickHouse connected to.
-   `port` ([String](../../sql-reference/data-types/string.md)) — The port of the ZooKeeper node that ClickHouse connected to.
-   `index` ([UInt8](../../sql-reference/data-types/int-uint.md)) — The index of the ZooKeeper node that ClickHouse connected to. The index is from ZooKeeper config.
-   `connected_time` ([String](../../sql-reference/data-types/string.md)) — When the connection was established
-   `is_expired` ([UInt8](../../sql-reference/data-types/int-uint.md)) — Is the current connection expired.
-   `keeper_api_version` ([String](../../sql-reference/data-types/string.md)) — Keeper API version.
-   `client_id` ([UInt64](../../sql-reference/data-types/int-uint.md)) — Session id of the connection.
```
